### PR TITLE
Exclude Goerli from the overview tab

### LINKED ISSUE #2125: Exclude Goerli from overview tab
Test nets shouldn't be included in the overview tab data.
Let's remove them from:
- total balance
- accounts balances
- networks chart
- assets balance

image

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes #2125 

This PR excludes Goerli from the overview tab. The Goerli network is no longer visible on the overview page. 

**UI**

Before

Screenshot 2022-09-07 at 10 00 1

After
Screenshot 2022-09-09 at 16 22 55

**To Test**

Go to the overview page and check if Goerli isn't visible on: 

- [x] total balance
- [x] accounts balances
- [x] networks chart
- [x] assets balance

### CODE REVIEW

File: background/constants/networks.ts [hunk 1]
```diff
@@ -98,3 +98,7 @@ export const NETWORK_BY_CHAIN_ID = {
   [KOVAN.chainID]: KOVAN,
   [FORK.chainID]: FORK,
 }
+
+export const TEST_NETWORK_BY_CHAIN_ID = {
```
Comment (0xDaedalus) [inline background/constants/networks.ts:102]: Lets use a `Set` instead - that should reduce the runtime complexity of the dependent code.

File: background/redux-slices/selectors/accountsSelectors.ts [hunk 2]
```diff
@@ -493,7 +501,23 @@ export const getAllNetworks = createSelector(getAccountState, (account) =>
...
+    ).length
+)
+
+export const getTotalBalanceForOverview = createSelector(
+  selectAccountsTotalForOverview,
+  (accountsTotal) =>
+    Object.values(accountsTotal)
+      .reduce(
+        (total, { totals }) =>
+          Object.values(totals).reduce((sum, value) => sum + value) + total,
```
Comment (0xDaedalus) [inline background/redux-slices/selectors/accountsSelectors.ts:518]: Can we find a better variable name than `value` here?

File: background/redux-slices/selectors/accountsSelectors.ts [hunk 3]
```diff
@@ -412,16 +416,20 @@ export type AccountTotalList = {
     }
   }
 }
-/** Get list of all accounts totals on all networks */
-export const selectAccountsTotal = createSelector(
+/** Get list of all accounts totals on all networks but without test networks */
+export const selectAccountsTotalForOverview = createSelector(
```
Comment (0xDaedalus) [inline background/redux-slices/selectors/accountsSelectors.ts:420]: `selectAccountTotalsForOverview`

File: background/redux-slices/selectors/accountsSelectors.ts [hunk 4]
```diff
@@ -493,7 +501,23 @@ export const getAllNetworks = createSelector(getAccountState, (account) =>
   )
 )
 
-export const getNetworkCount = createSelector(
+export const getNetworkCountForOverview = createSelector(
   getAllNetworks,
-  (allNetworks) => allNetworks.length
+  (allNetworks) =>
+    allNetworks.filter(
+      (name) => !Object.values(TEST_NETWORK_BY_CHAIN_ID).includes(name)
```
Comment (0xDaedalus) [inline background/redux-slices/selectors/accountsSelectors.ts:508]: I believe that `getAllNetworks` returns a list of `EVMNetworks`

File: background/redux-slices/selectors/accountsSelectors.ts (hunk 2 above)
Comment (kkosiorowska) [reply background/redux-slices/selectors/accountsSelectors.ts:518]: I will change the name from `value` to `balance`

File: background/redux-slices/selectors/accountsSelectors.ts (hunk 4 above)
Comment (kkosiorowska) [reply background/redux-slices/selectors/accountsSelectors.ts:508]: It will be updated when I use `Set` instead for `TEST_NETWORK_BY_CHAIN_ID`

**Review by 0xDaedalus — APPROVED** [review]
Tested and seems to be working great - type naming comment is non blocking but would love to see us follow up on it later - mind making an issue for it?

File: background/redux-slices/accounts.ts [hunk 5]
```diff
@@ -47,16 +48,18 @@ type AccountData = {
   defaultAvatar: string
 }
 
+type Evm = {
```
[the remark this answers is not in the document]
Comment (mhluongo) [reply background/redux-slices/accounts.ts:51]: > naming things is hard

[the remark this answers is not in the document]
Comment (jagodarybacka) [reply background/redux-slices/accounts.ts:51]: `AccountsByChainID`, `EVMAccounts` or something like that?

**Pull request merged**